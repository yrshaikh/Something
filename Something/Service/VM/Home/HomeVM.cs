using Service.VM.Common;
using Service.VM.Enums;

namespace Service.VM.Home
{
    public class HomeVM : BasePageVM
    {
        public override int PageTypeId { get; set; } = (int)PageTypeEnum.Home;
    }
}
