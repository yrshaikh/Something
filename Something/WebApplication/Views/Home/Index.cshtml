@using WebApplication.Utils;
@using Service.VM.Home;
@{
    ViewData["Title"] = "Index";
}

<div id="root"></div>

<script type="text/javascript">
    window.params = @Html.Raw(JsonSerializer.Serialize<HomeVM>(Model));
</script>

<script src="~/dist/home.bundle.js"></script>