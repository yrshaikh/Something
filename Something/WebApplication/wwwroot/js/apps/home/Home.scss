@import "../Global.scss";
@import "../Variables.scss";

body {
    background-color: #636381;
}

.Home {
    color:white;
    margin: 0 auto;
    width: 500px;
    margin-top: 15px;
    h1 {
        color: white;
        font-size: 48px;
        margin-bottom: 0;
    }
    p {
        font-size: 14px;
    }

    &__btns {
        &--default {
            padding: 8px 10px;
            font-size: 12px;
            width: 140px;
            margin-right: 10px;
            display: inline-block;
            border-radius: 50px;
            text-align: center;
            text-transform: uppercase;
        }
        &--yellow {
            background-color: #ffbc00;
            border: 1px solid #ffbc00;
            color: #000000;
        }
    }

    &__gallery {
        img {
            width: 500px;
            border-radius: 5px;
            margin-top: 10px;
        }
    }
}