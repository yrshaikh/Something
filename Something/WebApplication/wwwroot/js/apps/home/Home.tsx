import * as React from "react";
import { Button } from "antd";

import "./Home.scss";
import { IHomeProps } from "./types/IHomeProps";

export const Home : React.SFC<IHomeProps> = () => {
    return <div className="Home">
        <h1>Something.</h1>
        <p>An open source project and issue management app that doesn't suck!</p>
        <div className="Home__btns">
            <a
                className="Home__btns--default Home__btns--yellow"
                href="/account/login">
                Login
            </a>
            <a
                className="Home__btns--default Home__btns--yellow"
                href="/account/register">
                Register
            </a>
        </div>
        <div className="Home__gallery">
            <img src="https://cdn.dribbble.com/users/399068/screenshots/5374838/2018-10-10_155047.png" />
        </div>
    </div>;
}
