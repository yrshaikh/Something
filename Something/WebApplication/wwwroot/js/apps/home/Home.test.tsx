import "jest";
import * as React from "react";
import { shallow } from "enzyme";
import enzymeToJson from "enzyme-to-json";

import { Home } from "./Home";

describe("<Home />", () => {
    it("should render correctly", () => {
        const wrapper = shallow(<Home />);
        expect(enzymeToJson(wrapper)).toMatchSnapshot();
    });

    it("has expected contents on it", () => {
        const wrapper = shallow(<Home />);
        expect(wrapper.find('.Home').length).toBe(1);
        expect(wrapper.find('.Home').find('h1').length).toBe(1);
        expect(wrapper.find('.Home').find('p').length).toBe(1);
        
        const buttonWrapper = wrapper.find('.Home').find('.Home__btns');
        expect(buttonWrapper.length).toBe(1);
        const loginLink = buttonWrapper.find('.Home__btns--default').at(0);
        const registerLink = buttonWrapper.find('.Home__btns--default').at(1);
        expect(loginLink.prop('href')).toBe('/account/login');
        expect(registerLink.prop('href')).toBe('/account/register');
        
        const galleryWrapper = wrapper.find('.Home').find('.Home__gallery');
        expect(buttonWrapper.length).toBe(1);
        expect(galleryWrapper.find('img').length).toBe(1);
    });
});
