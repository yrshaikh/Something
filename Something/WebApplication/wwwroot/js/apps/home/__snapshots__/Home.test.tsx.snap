// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render correctly 1`] = `
<div
  className="Home"
>
  <h1>
    Something.
  </h1>
  <p>
    An open source project and issue management app that doesn't suck!
  </p>
  <div
    className="Home__btns"
  >
    <a
      className="Home__btns--default Home__btns--yellow"
      href="/account/login"
    >
      Login
    </a>
    <a
      className="Home__btns--default Home__btns--yellow"
      href="/account/register"
    >
      Register
    </a>
  </div>
  <div
    className="Home__gallery"
  >
    <img
      src="https://cdn.dribbble.com/users/399068/screenshots/5374838/2018-10-10_155047.png"
    />
  </div>
</div>
`;
